src/app/models/work-item.ts

```diff
@@ -104,4 +113,81 @@ export interface WorkItemUI {
   createdAt: string;
   labels: LabelUI[];
   comments: CommentUI[];
+  wiState: string;
+  wiDescription: string;
+}
+
+export class WorkItemMapper implements Mapper<WorkItemService, WorkItemUI> {
+  itMapper = new IterationMapper();
+  wiTypeMapper = new WorkItemTypeMapper();
+  areaMapper = new AreaMapper();
+  userMapper = new UserMapper();
+  lMapper = new LabelMapper();
+  cMapper = new CommentMapper();
+  
+  serviceToUiMapTree: MapTree = [{
+      fromPath: ['id'],
+      toPath: ['id']
+    }, {
+      fromPath: ['attributes','system.title'],
+      toPath: ['title']
+    }, {
+      fromPath: ['attributes','system.number'],
+      toPath: ['sysNumber']
+    }, {
+      fromPath: ['attributes','createdAt'],
+      toPath: ['createdAt']
+    }, 
+  ];
+
+  uiToServiceMapTree: MapTree = [{
+      toPath: ['id'],
+      fromPath: ['id']
+    }, {
+      toPath: ['attributes','system.title'],
+      fromPath: ['title']
+    }, {
+      toPath: ['type'],
+      toValue: 'workitems'
+    }, {
+      toPath: ['attributes','system.state'],
+      fromPath: ['wiState']
+    }
+  ];
+  
+  toUIModel(arg: WorkItemService): WorkItemUI {
+    let workItemUI = switchModel<WorkItemService, WorkItemUI>(
+      arg, this.serviceToUiMapTree
+    );
```

Comment:
This has been done so that the properties that are of base types can be mapped using the mapper